### DISCUSSION TITLE
custom Hooks

### DISCUSSION THREAD

**Discussion by honaseung** [honaseung/react-deep-dive#30] [Chapter3]
Till now, We've been studied about reserved 'hooks' of 'React'. React allows you making your own custom 'hooks'. Cutom hooks help you to integrate your codes. But you have to follow some rules.

1. custom hook's name should be start with 'use'. ex) useCustomStuff
2. custom hooks only can be imported from `React Component`.
3. custom hooks only can be called on the top of the `Component`.

These are you should be know when you use `custom hooks`. Simple right? You can use some function of `React`. Such like `useState` or `useEffect` or etc... except return `JSX`. But it doesn't affect to render. (Actually, So I wonder what is the point of using `React` function in custom hooks?) This is the way of `custom hooks` helping integrate codes of function.

**Comment by chaminju927** [comment]
Doesn't it affect rendering when using functions of React like useState or useEffect in custom Hooks?

**Reply by honaseung** [reply]
It affect rendering when its `state` is changed. What I meant was if `state` is changed insideof `customHook`, Then it will not be rendered.
